### PR TITLE
feat(taskform): error handling for group exceeding limit on tasks

### PR DESCRIPTION
**Pull request by il10**
Error handling for group exceeding limit on tasks

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @il10 has signed the CLA. Thanks for the pull request!

File: src/api/Feed.js [hunk 1]
```diff
@@ -459,6 +466,7 @@ class Feed extends Base {
                 successCallback();
             }
         } catch (e) {
+            // debugger;
```
Comment (il10) [inline src/api/Feed.js]: delete line.

File: src/api/Feed.js [hunk 2]
```diff
@@ -615,17 +623,63 @@ class Feed extends Base {
...
+        // Pull out group id from groupAssignees array
+        const groupIds = groupAssignees.map(assignee => assignee.id);
+
+        // run API call against each id --> returns an array of API calls (promises)
+        this.groupsAPI = new GroupsAPI(this.options);
+        const groupInfoPromises: Array<Promise<any>> = groupIds.map(groupId =>
+            this.groupsAPI.getGroupCount({
+                file,
+                group: { id: groupId },
+                successCallback: console.log,
```
Comment (il10) [inline src/api/Feed.js]: Leaving `successCallback` empty prevents the warning in the UI from displaying. It instead will exit out of the modal and the next time the user tries to create a task, the warning will be displayed then ¯\_(ツ)_/¯

File: src/api/Feed.js [hunk 3]
```diff
@@ -615,17 +623,63 @@ class Feed extends Base {
...
+                errorCallback: (e: ElementsXhrError, code: string) => {
+                    this.feedErrorCallback(true, e, code);
+                },
+            }),
+        );
+
+        // Fetch each group size in parallel --> return an array of group sizes
+        Promise.all(groupInfoPromises).then((groupCounts: Array<{ total_count: number }>) => {
+            const maxCount = 3;
```
Comment (il10) [inline src/api/Feed.js]: maxCount is set to 3 here for testing purposes. Final state is 250.

File: src/api/tasks/__tests__/TaskCollaborators.test.js [hunk 4]
```diff
@@ -75,6 +78,40 @@ describe('api/TaskCollaborators', () => {
             });
         });
 
+        describe('createTaskCollaboratorsforGroup()', () => {
```
Comment (il10) [inline src/api/tasks/__tests__/TaskCollaborators.test.js]: Adding a test for `createTaskCollaboratorsforGroup` as it was missed in a previous PR.

File: src/elements/content-sidebar/activity-feed/task-form/TaskError.js [hunk 5]
```diff
@@ -45,9 +52,22 @@ const TaskError = ({ editMode, error, taskType }: Props) => {
     }
 
     return (
-        <InlineNotice type="error" title={<FormattedMessage {...errorTitle} />}>
-            <FormattedMessage {...errorMessage} />
-        </InlineNotice>
+        <div>
+            {taskGroupExceedsError ? (
```
Comment (il10) [inline src/elements/content-sidebar/activity-feed/task-form/TaskError.js]: Warning for when groups exceeds limit

File: src/elements/content-sidebar/activity-feed/task-form/TaskForm.js [hunk 6]
```diff
@@ -349,6 +350,7 @@ class TaskForm extends React.Component<Props, State> {
         const isCompletionRuleCheckboxChecked = completionRule === TASK_COMPLETION_RULE_ANY;
         const isForbiddenErrorOnEdit = isLoading || (getProp(error, 'status') === 403 && !isCreateEditMode);
 
+        console.log('Task Error object: ', error);
```
Comment (il10) [inline src/elements/content-sidebar/activity-feed/task-form/TaskForm.js]: Remove.

File: src/elements/content-sidebar/activity-feed/task-new/Task.js [hunk 7]
```diff
@@ -153,6 +153,7 @@ class Task extends React.Component<Props, State> {
         }
     };
 
+    /* controls task error being shown */
```
Comment (il10) [inline src/elements/content-sidebar/activity-feed/task-new/Task.js]: Remove.

File: src/api/Feed.js [hunk 8]
```diff
@@ -615,17 +623,63 @@ class Feed extends Base {
...
+
+        // run API call against each id --> returns an array of API calls (promises)
+        this.groupsAPI = new GroupsAPI(this.options);
+        const groupInfoPromises: Array<Promise<any>> = groupIds.map(groupId =>
+            this.groupsAPI.getGroupCount({
+                file,
+                group: { id: groupId },
+                successCallback: console.log,
+                errorCallback: (e: ElementsXhrError, code: string) => {
+                    this.feedErrorCallback(true, e, code);
```
Comment (il10) [inline src/api/Feed.js]: first param should be `false`

File: src/api/Feed.js (hunk 2 above)
Comment (alexkrolick) [reply src/api/Feed.js]: Put a `noop` here?

**Comment by booc0mtaco** [comment]
will close in favor of a PR based on these chunks + some additional functionality.

**Pull request closed without merging**